core/src/test/scala/zio/config/SetTest.scala

```diff
@@ -271,6 +271,113 @@ object SetTest
             )
           )
           assert(res)(isLeft(equalTo(expected)))
+        },
+        test("accumulates all errors") {
+          case class CfgA(a1: Boolean, a2: Int)
+          case class Cfg(a: Set[CfgA], b: Set[Int])
+
+          val cCfgA = (boolean("a1") |@| int("a2"))(CfgA, CfgA.unapply)
+          val cCfg  = (nested("a")(setStrict(cCfgA)) |@| nested("b")(setStrict(int)))(Cfg, Cfg.unapply)
+
+          val res = read(
+            cCfg from ConfigSource.fromPropertyTree(
+              Record(
+                Map(
+                  "a" -> PropertyTree.Sequence(
+                    List(
+                      Record(
+                        Map("a1" -> Leaf("true"), "a2" -> Leaf("lorem ipsum"))
+                      ),
+                      Record(
+                        Map("a1" -> Leaf("true"), "a2" -> Leaf("1"))
+                      ),
+                      Record(
+                        Map("a1" -> Leaf("false"))
+                      ),
+                      Record(
+                        Map("a2" -> Leaf("2"))
+                      )
+                    )
+                  ),
+                  "b" -> Sequence(Leaf("2") :: Leaf("one") :: Leaf("2") :: Nil)
+                )
+              ),
+              "tree"
+            )
+          )
+          val expected: ReadError[String] =
+            AndErrors(
+              List(
+                AndErrors(
+                  List(
+                    MissingValue(List(Key("a"), Index(3), Key("a1"))),
+                    MissingValue(List(Key("a"), Index(2), Key("a2"))),
+                    FormatError(
+                      List(Key("a"), Index(0), Key("a2")),
+                      "Provided value is lorem ipsum, expecting the type int"
+                    )
+                  )
+                ),
+                AndErrors(
+                  List(
+                    FormatError(List(Key("b"), Index(1)), "Provided value is one, expecting the type int")
+                  )
+                )
+              )
+            )
+          assert(res)(isLeft(hasField("size", _.size, equalTo(expected.size))))
```

Comment:
I think that `ConversionError(List(Key("b"), Index(0)), "Duplicated values found")` error is missing for that case. Am I right?

Reply:
@marekklis  I am not sure. Is it a bug you talking about?

Reply:
@afsalthaj yes, I think this is a bug. If `b` is a set of int, for the following input:
```
"b" -> Sequence(Leaf("2") :: Leaf("one") :: Leaf("2") :: Nil)
```
we should get two errors accumulated:
`FormatError(..., "Provided value is one, expecting the type int")` and `ConversionError(..., "Duplicated values found")` but we there is only `FormatError` returned.

Reply:
@marekklis Sorry for late response. Sure, is this only in the case of Set that we have this corner case? I guess the issue is it is not able to parse all the individual elements first to then “Try” forming a set. I think, it’s a bug, but it is ok to merge this current PR once you fix the build issue.
Let’s keep track of the expectation with set as a separate issue.